storage/federated/ha_federated.cc

```diff
@@ -2054,31 +2107,38 @@ int ha_federated::update_row(const uchar *old_data, uchar *) {
     }
 
     if (bitmap_is_set(table->read_set, (*field)->field_index())) {
-      size_t field_name_length = strlen((*field)->field_name);
-      append_ident(&where_string, (*field)->field_name, field_name_length,
-                   ident_quote_char);
-      if ((*field)->is_null_in_record(old_data))
-        where_string.append(STRING_WITH_LEN(" IS NULL "));
-      else {
-        bool needs_quote = (*field)->str_needs_quotes();
-        where_string.append(STRING_WITH_LEN(" = "));
+      if ((*field)->type() == MYSQL_TYPE_DOUBLE &&
```

Comment:
Do we need to do the same for `MYSQL_TYPE_FLOAT`?
Also `MYSQL_TYPE_DECIMAL` and `MYSQL_TYPE_NEWDECIMAL`